src/User/UserServiceProvider.php

```diff
@@ -92,9 +127,12 @@ public function boot()
         $events->listen(EmailChangeRequested::class, EmailConfirmationMailer::class);
 
         $events->subscribe(UserMetadataUpdater::class);
-        $events->subscribe(UserPolicy::class);
 
         $events->listen(ConfigureUserPreferences::class, [$this, 'configureUserPreferences']);
+
+        foreach ($this->app->make('flarum.policies') as $policy) {
+            $events->listen(ScopeModelVisibility::class, [$this->app->make($policy), 'scopeQueryListener']);
```

Comment:
same here, let's resolve all policies once in an array before making the gate and use that.